cpg-core/src/main/java/de/fraunhofer/aisec/cpg/passes/RegisterDependency.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.passes
+
+import kotlin.reflect.KClass
+
+/**
+ * Register a dependency for the annotated pass. This ensures that:
+ * - the annotated pass is executed after its dependency when `softDependency` is `false`:
+ * - the dependency is added to the list of active passes even if not manually specified by the user
+ * when `softDependency`is true:
+ * - the dependency is not added to the list of active passes - the order is only enforced if the
+ * user manually adds the pass
+ */
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.CLASS)
+@Repeatable
+annotation class RegisterDependency(
```

Comment:
My suggestion was just a simple `DependsOn` or do you prefer the `RegisterDependency` name? I thin annotations should probably describe a "state" rather than an "action".